app/controllers/routeDetails/RedirectController.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.routeDetails
+
+import com.google.inject.Inject
+import controllers.actions.Actions
+import models.{LocalReferenceNumber, NormalMode}
+import navigation.UserAnswersNavigator
+import navigation.routeDetails.RouteDetailsNavigatorProvider
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.CountriesService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import scala.concurrent.ExecutionContext
+
+class RedirectController @Inject() (
+  override val messagesApi: MessagesApi,
+  actions: Actions,
+  val controllerComponents: MessagesControllerComponents,
+  navigatorProvider: RouteDetailsNavigatorProvider,
+  countriesService: CountriesService
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def redirect(lrn: LocalReferenceNumber): Action[AnyContent] = actions.requireData(lrn).async {
+    implicit request =>
+      for {
+        ctcCountries                          <- countriesService.getCountryCodesCTC()
+        customsSecurityAgreementAreaCountries <- countriesService.getCustomsSecurityAgreementAreaCountries()
+      } yield {
```

Comment:
Is there any way we can avoid this as these endpoints are now frequently getting called on most route pages + submission pages? I have made a a note of this on the refactor confluence page however was wondering if there was possibly a quick solution: https://confluence.tools.tax.service.gov.uk/display/CTCTA/CTC+Traders+Web+code+base+refactoring

Reply:
@MatthewAlexanderBeck so these were currently getting called anyway as part of the navigator provider initialisation but I've moved that logic out so each controller makes the calls once and shares the results between the navigate() and updateTask() steps

Reply:
Happy to refactor these at a later date, maybe even move the calls into the action builder, but wonder how efficient it is to store really quite sizeable reference data lists inside each user's mongo document?